File: FinanceApp.Application/Dtos/TransactionDtos/CreateTransactionDto.cs
using FinanceApp.Domain.Entities;
using Microsoft.AspNetCore.Http;

namespace FinanceApp.Application.Dtos.TransactionDtos;

public class CreateTransactionDto
{
  public string Name { get; set; } = string.Empty;
  public string? Description { get; set; }
  public Money Value { get; set; } = new();
  public DateTimeOffset TransactionDate { get; set; }
  public TransactionTypeEnum TransactionType { get; set; }
  public string? TransactionGroupId { get; set; } = null!;
}

File: FinanceApp.Web/src/models/TransactionDtos/CreateTransactionDto.ts
import { TransactionTypeEnum } from "../Enums/TransactionType.enum";
import { Money } from "../Money/Money";

export interface CreateTransactionDto {
    name: string,
    description: string,
    value: Money,
    transactionDate: Date,
    transactionType: TransactionTypeEnum,
    transactionGroupId?: string
}
